patches/6807.diff

```diff
@@ -0,0 +1,37 @@
+diff --git a/extide/gradle/src/org/netbeans/modules/gradle/api/execute/GradleDistributionManager.java b/extide/gradle/src/org/netbeans/modules/gradle/api/execute/GradleDistributionManager.java
+index 1cc156d0f622..c90316077493 100644
+--- a/extide/gradle/src/org/netbeans/modules/gradle/api/execute/GradleDistributionManager.java
++++ b/extide/gradle/src/org/netbeans/modules/gradle/api/execute/GradleDistributionManager.java
+@@ -100,6 +100,7 @@ public final class GradleDistributionManager {
+         GradleVersion.version("7.5"), // JDK-18
+         GradleVersion.version("7.6"), // JDK-19
+         GradleVersion.version("8.3"), // JDK-20
++        GradleVersion.version("8.5"), // JDK-21
+     };
+ 
+     final File gradleUserHome;
+@@ -484,12 +485,22 @@ public String getVersion() {
+          * Checks if this Gradle distribution is compatible with the given
+          * major version of Java. Java 1.6, 1.7 and 1.8 are treated as major
+          * version 6, 7, and 8.
+-         *
++         * <p>
++         * NetBeans uses a built in fixed list of compatibility matrix. That
++         * means it might not know about the compatibility of newer Gradle
++         * versions. Optimistic bias would return {@code true} on these
++         * versions form 2.37. 
++         * </p>
+          * @param jdkMajorVersion the major version of the JDK
+          * @return <code>true</code> if this version is supported with that JDK.
+          */
+         public boolean isCompatibleWithJava(int jdkMajorVersion) {
+-            return jdkMajorVersion <= lastSupportedJava();
++            
++            GradleVersion lastKnown = JDK_COMPAT[JDK_COMPAT.length - 1];
++            // Optimistic bias, if the GradleVersion is newer than the last NB
++            // knows, we say it's compatible with any JDK
++            return lastKnown.compareTo(version.getBaseVersion()) < 0 
++                    || jdkMajorVersion <= lastSupportedJava();
```

Comment:
can we make use of  Gradle compatibility table given here https://docs.gradle.org/current/userguide/compatibility.html ?

Reply:
Yes maybe I don't know for sure, but currently it is manually handled in netbeans for every release. This patch is taken from https://github.com/apache/netbeans/pull/6807